### PR TITLE
fix(windows): coverity reports for mcompile

### PR DESCRIPTION
**Pull request by mcdurdin**

### CODE REVIEW

File: windows/src/engine/mcompile/mc_unicode.cpp [hunk 1]
```diff
@@ -261,11 +261,11 @@ const WCHAR cp1252[256] = {
 void InplaceUnicode(PWSTR p) {
   while(p && *p) {
     if(*p != UC_SENTINEL) {
-      if(*p >= 0 && *p <= 0xFF) {
```
Comment (jahorton) [inline windows/src/engine/mcompile/mc_unicode.cpp:264]: Was the `>=0` check redundant here?  Otherwise, why the change?

File: windows/src/engine/mcompile/mc_unicode.cpp [hunk 2]
```diff
@@ -276,7 +276,7 @@ BOOL ConvertKeyboardToUnicode(LPKEYBOARD kbd) {
   LPKEY kp;
   DWORD i, j;
 
-  if(kbd->StartGroup[BEGIN_UNICODE] >= 0 && kbd->StartGroup[BEGIN_UNICODE] < kbd->cxGroupArray) {
```
Comment (jahorton) [inline windows/src/engine/mcompile/mc_unicode.cpp:279]: So... I assume this means that `kbd->StartGroup[BEGIN_UNICODE]` is set to -1 or something like that when it is not available?  I'd normally expect a `0` for a null state, though I understand if that's not the case here for legacy reasons.

File: windows/src/engine/mcompile/mc_unicode.cpp (hunk 1 above)
Comment (mcdurdin) [reply windows/src/engine/mcompile/mc_unicode.cpp:264]: `PWSTR` is a pointer to unsigned words, so `>= 0` is always true. Coverity picks it up because it suggests that we might have a logic problem.

File: windows/src/engine/mcompile/mc_unicode.cpp (hunk 2 above)
Comment (mcdurdin) [reply windows/src/engine/mcompile/mc_unicode.cpp:279]: `kbd->StartGroup[BEGIN_UNICODE]` are unsigned long, so set to `0xFFFFFFFFL` (`-1`) on line 461 of Compiler.cpp. The `>= 0` test is redundant because of unsigned. This is an index into an array, so `-1` is equivalent to null state.

Comment (jahorton) [reply windows/src/engine/mcompile/mc_unicode.cpp:279]: So the remaining check still covers the `null` case?  It's not immediately clear that `kbd->StartGroup[BEGIN_UNICODE] == 0xFFFFFFFFL` can only be the case if `kbd->cxGroupArray` also holds that value, which is the one way I can see the "`null`" case still being handled here.

Comment (mcdurdin) [reply windows/src/engine/mcompile/mc_unicode.cpp:279]: This is testing that `kbd->StartGroup` is in a valid range (`0 .. kbd->cxGroupArray-1`), and `0xFFFFFFFFL` is always going to be outside the valid range.

**Review by jahorton — APPROVED** [review]
